TKGame/Components/Interface/LevelComponent.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TKGame.Level_Editor_Content;
+
+namespace TKGame.Components.Interface
+{
+    public interface LevelComponent
+    {
+        virtual public void LevelComponent() { }
+        abstract internal void Update();
+        abstract internal void AddLevel(Level newLevel);
+        abstract internal Level GetCurrentLevel();
+        abstract internal Stage GetCurrentStage();
+        abstract internal Stage GetPreviousStage();
+        abstract internal Stage GetNextStage();
+        abstract internal bool IsCurrentStageFirst();
+        abstract internal bool IsCurrentStageFinal();
+        abstract internal void SetCurrentStage(Stage stage);
+        abstract internal void GoToNextStage();
+        abstract internal void GoToPreviousStage();
```

Comment:
Is there a reason to have these be marked as `virtual` and `abstract` instead of just leaving them just be like `void Update();`? Any object that inherits this interface will have to override these functions anyways. Also `internal` is a bit pointless since we are only working on one assembly, so it's the same as `public`